File: network/src/main/java/com/example/network/utils/DateConverter.java
package com.example.network.utils;

import com.google.gson.JsonDeserializationContext;
import com.google.gson.JsonDeserializer;
import com.google.gson.JsonElement;
import com.google.gson.JsonParseException;
import com.google.gson.JsonPrimitive;
import com.google.gson.JsonSerializationContext;
import com.google.gson.JsonSerializer;

import java.lang.reflect.Type;
import java.util.Date;

public class DateConverter implements JsonDeserializer<Date>, JsonSerializer<Date> {

    @Override
    public Date deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context) throws JsonParseException {
        return new Date(json.getAsLong());
    }

    @Override
    public JsonElement serialize(Date src, Type typeOfSrc, JsonSerializationContext context) {
        return new JsonPrimitive(src.getTime());
    }
}

File: network/src/main/java/com/example/network/utils/DateConverter.kt
package com.example.network.utils

import com.google.gson.*
import java.lang.reflect.Type
import java.util.*

class DateConverter : JsonDeserializer<Date>, JsonSerializer<Date> {

//    throws JsonParseException
    override fun  deserialize(json: JsonElement,  typeOfT:Type,  context: JsonDeserializationContext):Date  {
        return  Date(json.asLong)
    }

    override fun  serialize( src: Date,  typeOfSrc:Type,  context: JsonSerializationContext):JsonElement {
        return  JsonPrimitive(src.time)
    }
}
